src/definition/accessors/IExternalComponentsExtend.ts

```diff
@@ -0,0 +1,9 @@
+import { IExternalComponent } from '../externalComponent';
+
+/**
+ * This accessor provides methods for register an external
+ * component. Provided during the initialization of your App.
+ */
+export interface IExternalComponentsExtend {
+    register(externalComponent: IExternalComponent): Promise<void>;
```

Comment:
Please add documentation to this method and see my other comment on this class's implementation.

Reply:
resolved in the PR: https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/156